src/core/config/link_metrics_manager.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) 2023, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes compile-time configurations for Link Metrics Manager.
+ *
+ */
+
+#ifndef CONFIG_LINK_METRICS_MANAGER_H_
+#define CONFIG_LINK_METRICS_MANAGER_H_
+
+/**
+ * @def OPENTHREAD_CONFIG_LINK_METRICS_MANAGER_ENABLE
+ *
+ * Define as 1 to enable Link Metrics Manager support.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_LINK_METRICS_MANAGER_ENABLE
+#define OPENTHREAD_CONFIG_LINK_METRICS_MANAGER_ENABLE 0
```

Comment:
There's an increasement of about 4K (1 for text and 3 for bss) in total for FTD, and an increasement of about 8K for MTD (because to enable this, `LINK_METRICS_INITIATOR` is also enabled). So I guess we can just enable this on FTD/BR and disable on MTD.

Reply:
Screenshot the size report

![Screenshot 2023-08-30 at 11 47 55](https://github.com/openthread/openthread/assets/12945188/7d6cf637-f43c-4f27-843e-21de9d41f6d7)